src/concurrency.md

```diff
@@ -70,7 +71,102 @@ fn main() {
 }
 ```
 
+[ex-rayon-thumbnails]: #ex-rayon-thumbnails
+<a name="ex-rayon-thumbnails"></a>
+## Generate thumbnails for all .jpg in current directory
 
+[![rayon-badge]][rayon] [![glob-badge]][glob] [![image-badge]][image] [![cat-concurrency-badge]][cat-concurrency] [![cat-filesystem-badge]][cat-filesystem]
+
+This example uses the `glob`, `image`, and `rayon` crates to generate thumbnails for all .jpg files in the current directory. 
+The thumbnails are stored in a new folder called `thumbnails`.
+
+Files are found with [`glob::glob_with`] so we can match case insensitively on both `.jpg` and `.JPG`.
+Images are then resized in parallel with [`par_iter`] using [`DynamicImage::resize`].
+
+
+```rust
+extern crate glob;
+extern crate image;
+extern crate rayon;
+
+use std::path::{Path, PathBuf};
+use std::fs;
+use std::fs::File;
+
+use glob::{glob_with, MatchOptions};
+use image::{FilterType, ImageResult};
+use rayon::prelude::*;
+
+fn main() {
+    let files = find_jpg();
+    println!("Found {} files to convert", files.len());
+    
+    for problem in parallel_resize(&files, Path::new("thumbnails"), 300) {
+        println!("{:?} failed with error: {}", problem.file_path, problem.err);
+    }
+}
+
+/// Find all files that have a `.jpg` extension in the current directory
+fn find_jpg() -> Vec<PathBuf> {
+    let mut paths = Vec::new();
+
+    // use `glob_with()` and MatchOptions so we can match case insensitively to .jpg
+    let options = MatchOptions {
+        case_sensitive: false,
+        require_literal_separator: false,
+        require_literal_leading_dot: false,
+    };
+
+    for entry in glob_with("*.jpg", &options).expect("Failed to read glob pattern") {
+        match entry {
+            Ok(path) => paths.push(path),
+            Err(e) => println!("Error with: {:?}", e),
+        }
+    }
+
+    paths
+}
+
+/// Resize `file` to have a maximum dimension of `longest_edge` and save the resized
+/// image to the `thumb_dir` folder
+fn resize_jpg(file: &Path, thumb_dir: &Path, longest_edge: u32) -> ImageResult<()> {
+    // create the folder if it doesn't already exist
+    fs::create_dir_all(thumb_dir)?;
```

Comment:
please call the `create_dir_all` only once prior to multithreaded code. These have few nasty pitfals when used concurrently https://doc.rust-lang.org/std/fs/fn.create_dir_all.html#errors

Reply:
Yeah, that was left over from some other experiments I was doing. I think I originally did it in case of going into to multiple folders, but never got that far, so it was never tested... 

I'll change it over so it only does it once